aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -194,16 +292,26 @@ private static ApiCategory mockApiCategoryWithGraphQlApi() throws AmplifyExcepti
         doAnswer(invocation -> {
             int indexOfResponseConsumer = 1;
             Consumer<GraphQLResponse<String>> onResponse = invocation.getArgument(indexOfResponseConsumer);
-            onResponse.accept(new GraphQLResponse<>("{}", Collections.emptyList()));
+            // Calling onResponse with an invalid input generates an error in MutationOutbox.hasPendingMutation.
+            // Disabling it for now since it does not affect the assertions we need for this test.
+            // onResponse.accept(new GraphQLResponse<>("{}", Collections.emptyList()));
             return null;
         }).when(mockApiPlugin).mutate(any(GraphQLRequest.class), any(Consumer.class), any(Consumer.class));
 
         // Make believe that subscriptions return response immediately
         doAnswer(invocation -> {
-            int indexOfStartConsumer = 2;
-            Consumer<String> onResponse = invocation.getArgument(indexOfStartConsumer);
-            onResponse.accept(RandomString.string());
-            return null;
+            int indexOfStartConsumer = 1;
+            Consumer<String> onStart = invocation.getArgument(indexOfStartConsumer);
+            GraphQLOperation<?> mockOperation = mock(GraphQLOperation.class);
+            doAnswer(opAnswer -> {
+                this.subscriptionCancelledCounter.incrementAndGet();
+                return null;
+            }).when(mockOperation).cancel();
```

Comment:
Oh fuck, a `doAnswer` inside of a `doAnswer`, Head just exploded.

The situation with the `Cancelable` return type and `String` "on started" token is such a nightmare, isn't it? On a brighter day, maybe we could cleanup that API signature so we don't have to get so clever about managing/mocking/testing it.